### PR TITLE
Clarify limitations around capturing of parameters that have ref-like type

### PR DESCRIPTION
**Pull request by AlekseyTs**

### CODE REVIEW

File: proposals/primary-constructors.md [hunk 1]
```diff
@@ -151,7 +151,7 @@ All instance member initializers in the class body will become assignments in th
 
 If a primary constructor parameter is referenced from within an instance member, and the reference is not inside of a `nameof` argument, it is captured into the state of the enclosing type, so that it remains accessible after the termination of the constructor. A likely implementation strategy is via a private field using a mangled name. In a readonly struct the capture fields will be readonly. Therefore, access to captured parameters of a readonly struct will have similar restrictions as access to readonly fields. Access to captured parameters within a readonly member will have similar restrictions as access to instance fields in the same context.
 
-Capturing is not allowed for `ref`, `in` and `out` parameters. This is similar to a limitation for capturing in lambdas. 
+Capturing is not allowed for parameters that have ref-like type, `ref`, `in` and `out` parameters. This is similar to a limitation for capturing in lambdas.
```
Comment (cston) [inline proposals/primary-constructors.md:154]: Perhaps
```suggestion
Capturing is not allowed for parameters that have ref-like type, and capturing is not allowed for `ref`, `in` and `out` parameters. This is similar to a limitation for capturing in lambdas. 
```

File: proposals/primary-constructors.md [hunk 2]
```diff
@@ -151,7 +151,7 @@ All instance member initializers in the class body will become assignments in th
 
 If a primary constructor parameter is referenced from within an instance member, and the reference is not inside of a `nameof` argument, it is captured into the state of the enclosing type, so that it remains accessible after the termination of the constructor. A likely implementation strategy is via a private field using a mangled name. In a readonly struct the capture fields will be readonly. Therefore, access to captured parameters of a readonly struct will have similar restrictions as access to readonly fields. Access to captured parameters within a readonly member will have similar restrictions as access to instance fields in the same context.
 
-Capturing is not allowed for `ref`, `in` and `out` parameters. This is similar to a limitation for capturing in lambdas. 
+Capturing is not allowed for parameters that have ref-like type, and capturing is not allowed for `ref`, `in` and `out` parameters. This is similar to a limitation for capturing in lambdas.
```
Comment (AlekseyTs) [inline proposals/primary-constructors.md:154]:
```suggestion
Capturing is not allowed for parameters that have ref-like type, and capturing is not allowed for `ref`, `in` or `out` parameters. This is similar to a limitation for capturing in lambdas. 
```

**Review by cston — APPROVED** [review]

File: proposals/primary-constructors.md [hunk 3]
```diff
@@ -151,7 +151,7 @@ All instance member initializers in the class body will become assignments in th
 
 If a primary constructor parameter is referenced from within an instance member, and the reference is not inside of a `nameof` argument, it is captured into the state of the enclosing type, so that it remains accessible after the termination of the constructor. A likely implementation strategy is via a private field using a mangled name. In a readonly struct the capture fields will be readonly. Therefore, access to captured parameters of a readonly struct will have similar restrictions as access to readonly fields. Access to captured parameters within a readonly member will have similar restrictions as access to instance fields in the same context.
 
-Capturing is not allowed for `ref`, `in` and `out` parameters. This is similar to a limitation for capturing in lambdas. 
+Capturing is not allowed for parameters that have ref-like type, and capturing is not allowed for `ref`, `in` or `out` parameters. This is similar to a limitation for capturing in lambdas.
```
Comment (jaredpar) [inline proposals/primary-constructors.md:154]: Is this true even when the containing type is a `ref struct`? For example I don't _think_ the following presents any problem:

```csharp
ref struct S1 { }
ref struct S2(S1 s) {
  void M() {
    s.ToString();
  }
}
```

Comment (AlekseyTs) [reply proposals/primary-constructors.md:154]: > Is this true even when the containing type is a `ref struct`?

Yes. The current proposal doesn't have special rules for capturing in `ref struct`s.

**Pull request merged**